File: day01/solution.hs
import System.IO

readInt :: String -> Int
readInt = read

main :: IO()
main = do
    contents <- openFile "input.txt" ReadMode >>= hGetContents
    let nums = map readInt $ lines contents
    
    let countLatterGreater = sum . map (fromEnum . uncurry (<))

    -- Part 1
    print $ countLatterGreater $ zip nums (drop 1 nums)
    
    -- Part 2
    print $ countLatterGreater $ zip nums (drop 3 nums)

File: day01/solution.rb
input = File.open("input.txt").readlines.map(&:to_i)

# Part 1
p input.each_cons(2).count{ |a,b| b > a }

# Part 2
p input.each_cons(3).map(&:sum).each_cons(2).count{ |a,b| b > a }
